### PR TITLE
Remove some cli parameters

### PR DESCRIPTION
**Pull request by changweige**
Nydus-snapshotter now has its own toml based configuration file whose entries are duplicated with CLI startup parameters.
In the first phase, we will remove those not frequently used cli parameters and suggest users to migrate the configuration to toml based configuration file.

### CODE REVIEW

File: cmd/containerd-nydus-grpc/pkg/command/flags.go [hunk 1]
```diff
@@ -127,31 +93,13 @@ func buildFlags(args *Args) []cli.Flag {
 			Usage:       "whether to enable nydus-overlayfs",
 			Destination: &args.EnableNydusOverlayFS,
 		},
-		&cli.BoolFlag{
-			Name:        "enable-stargz",
```
Comment (imeoer) [inline cmd/containerd-nydus-grpc/pkg/command/flags.go:131]: Does this impact compatibility?

File: misc/snapshotter/entrypoint.sh [hunk 2]
```diff
@@ -17,10 +17,7 @@ if [ "$#" -eq 0 ]; then
 		--root ${NYDUS_LIB} \
 		--address ${NYDUS_RUN}/containerd-nydus-grpc.sock \
 		--log-level ${LEVEL} \
-		--enable-metrics=${ENABLE_METRICS} \
-		--enable-nydus-overlayfs=${ENABLE_NYDUS_OVERLAY} \
 		--daemon-mode ${NYDUSD_DAEMON_MODE} \
```
Comment (imeoer) [inline misc/snapshotter/entrypoint.sh:20]: Can we also remove `--daemon-mode` ? It seems using `shared`.

File: cmd/containerd-nydus-grpc/pkg/command/flags.go (hunk 1 above)
Comment (changweige) [reply cmd/containerd-nydus-grpc/pkg/command/flags.go:131]: I suppose it doesn't. Stargz compatibility does not work for a long time.

File: misc/snapshotter/entrypoint.sh (hunk 2 above)
Comment (changweige) [reply misc/snapshotter/entrypoint.sh:20]: `multiple` mode is the default one, so we should keep it?

File: cmd/containerd-nydus-grpc/pkg/command/flags.go (hunk 1 above)
Comment (imeoer) [reply cmd/containerd-nydus-grpc/pkg/command/flags.go:131]: Should other flags still be reserved to ensure that snapshotter can be started if users use old flags?

File: misc/snapshotter/entrypoint.sh (hunk 2 above)
Comment (imeoer) [reply misc/snapshotter/entrypoint.sh:20]: Ok, just keep `--daemon-mode` flag and use `multiple` by default.

File: cmd/containerd-nydus-grpc/pkg/command/flags.go (hunk 1 above)
Comment (changweige) [reply cmd/containerd-nydus-grpc/pkg/command/flags.go:131]: I am inclined not to reserve other parameters. We have to migrate to the configuration file pattern in the end.
This PR only removes the parameters that are not likely to be used.

File: misc/snapshotter/entrypoint.sh (hunk 2 above)
Comment (changweige) [reply misc/snapshotter/entrypoint.sh:20]: Let's change the shell variable default value to `multiple`

Comment (changweige) [reply misc/snapshotter/entrypoint.sh:20]: done

File: cmd/containerd-nydus-grpc/pkg/command/flags.go (hunk 1 above)
Comment (imeoer) [reply cmd/containerd-nydus-grpc/pkg/command/flags.go:131]: Ok, we need notice users that this is a breaking change in the release note.

**Review by imeoer — APPROVED** [review]

**Pull request merged**